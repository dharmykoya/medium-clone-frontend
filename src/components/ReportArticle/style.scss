.report-textarea {
  width: 100%;
  margin: 1rem auto;
  color: inherit;
}

.report-btn {
  width: 100px;
  height: 45px;
}

.report-btn-close {
  font-weight: 500;
  margin-left: 1.5rem;
  font-size: 18px;
  border-radius: 5px;
}

.report-btn-close:hover {
  color: inherit;
}
