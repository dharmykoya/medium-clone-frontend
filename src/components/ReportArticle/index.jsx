import React, { Component } from 'react';
import Textarea from '../TextArea/index.jsx';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './style.scss';

class ReportArticleForm extends Component {
  state = {
    reason: ''
  };

  handleSubmit = event => {
    event.preventDefault();
    const { reportArticleRequest, slug, token, handleClose } = this.props;
    const reason = { reason: this.state.reason };
    reportArticleRequest(slug, reason, token);
    this.setState({ reason: '' });
    handleClose();
  };

  handleInputChange = event => {
    this.setState({ reason: event.target.value });
  };

  render() {
    const { lightTheme, handleClose } = this.props;
    return (
      <form onSubmit={this.handleSubmit}>
        <h5>
          Help us understand the problem. What is going on with this Article?
        </h5>

        <Textarea
          customClassName="report-textarea"
          lightTheme={lightTheme}
          placeholder="Reason"
          value={this.state.reason}
          handleChange={this.handleInputChange}
          required
          name="reason"
        />
        <button className="button button-normal report-btn">Report</button>
        <Link
          className="btn button button-inverse report-btn report-btn-close"
          to="#"
          onClick={handleClose}
        >
          Close
        </Link>
      </form>
    );
  }
}

ReportArticleForm.propTypes = {
  lightTheme: PropTypes.bool,
  handleClose: PropTypes.func,
  reportArticleRequest: PropTypes.func,
  slug: PropTypes.string,
  token: PropTypes.string
};

export default ReportArticleForm;
