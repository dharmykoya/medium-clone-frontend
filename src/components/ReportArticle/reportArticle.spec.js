import React from 'react';
import { BrowserRouter } from 'react-router-dom';

import { mount } from 'enzyme';

import ReportArticle from './index.jsx';

describe('Report article form ', () => {
  const handleClose = jest.fn();
  const reportArticleRequest = jest.fn();
  const renderReportArticle = () => {
    const props = {
      lightTheme: false,
      slug: 'somerandomslug',
      token: 'somerandomtoken',
      handleClose,
      reportArticleRequest
    };
    return mount(
      <BrowserRouter>
        <ReportArticle {...props} />
      </BrowserRouter>
    );
  };

  it('should render report article successfully', () => {
    const event = { target: { name: 'report', value: 'spam' } };

    const wrapper = renderReportArticle();

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('textarea').simulate('change', event)).toBeTruthy();
  });

  it('should trigger handle close function', () => {
    const event = { target: { name: 'report', value: 'spam' } };

    const wrapper = renderReportArticle();

    expect(wrapper.find('Link').simulate('click', event)).toBeTruthy();
    expect(handleClose).toHaveBeenCalled();
  });

  it('should trigger handle close function', () => {
    const event = { target: { name: 'report', value: 'spam' } };

    const wrapper = renderReportArticle();

    expect(wrapper.find('button').simulate('click', event)).toBeTruthy();
  });

  it('should trigger onsubmit function', () => {
    const wrapper = renderReportArticle();

    expect(wrapper.find('button').simulate('submit')).toBeTruthy();
  });
});
