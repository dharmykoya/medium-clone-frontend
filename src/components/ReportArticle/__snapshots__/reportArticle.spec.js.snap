// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Report article form  should render report article successfully 1`] = `ReactWrapper {}`;
